weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+    tiff_location: str
+    use_personal_account: bool
+
+    @classmethod
+    def add_parser_arguments(cls, subparser: argparse.ArgumentParser):
+        subparser.add_argument('--xarray_open_dataset_kwargs', type=json.loads, default='{}',
+                               help='Keyword-args to pass into `xarray.open_dataset()` in the form of a JSON string.')
+        subparser.add_argument('--disable_in_memory_copy', action='store_true', default=False,
+                               help='To disable in-memory copying of dataset. Default: False')
+        subparser.add_argument('--disable_grib_schema_normalization', action='store_true', default=False,
+                               help='To disable merge of grib datasets. Default: False')
+        subparser.add_argument('--ee_asset', type=str, default=None,
+                               help='The asset folder path in earth engine project where the tiff image files'
+                               ' will be pushed.')
+        subparser.add_argument('-s', '--skip-region-validation', action='store_true', default=False,
+                               help='Skip validation of regions for data migration. Default: off')
+        subparser.add_argument('--tiff_location', type=str, default=None,
+                               help='The GCS location where the tiff files will be pushed.')
+        subparser.add_argument('-u', '--use_personal_account', action='store_true', default=False,
+                               help='To use personal account for earth engine authentication.')
+
+    @classmethod
+    def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.List[str]) -> None:
+        pipeline_options = PipelineOptions(pipeline_args)
+        pipeline_options_dict = pipeline_options.get_all_options()
+
+        if not known_args.tiff_location:
+            raise RuntimeError("'--tiff_location' is required for ee ingestion.")
+        if not known_args.ee_asset:
+            raise RuntimeError("'--ee_asset' is required for ee ingestion.")
+        if not re.match("^projects/.+/assets.*", known_args.ee_asset):
+            raise RuntimeError("'--ee_asset' is required to be in format: projects/+/assets/*.")
+
+        # Check that Cloud resource regions are consistent.
+        if not (known_args.dry_run or known_args.skip_region_validation):
+            # Program execution will terminate on failure of region validation.
+            logger.info('Validating regions for data migration. This might take a few seconds...')
+            validate_region(temp_location=pipeline_options_dict.get('temp_location'),
+                            region=pipeline_options_dict.get('region'))
+            logger.info('Region validation completed successfully.')
+
+    def __post_init__(self):
+        """Initializes Sink by initialize earthengine based on user input."""
+
+        if self.dry_run:
+            logger.debug('Uploaded assets into earth engine ...')
+            return
```

Comment:
Is this method necessary? I don't think this is a good place to use the dry run.

Reply:
Right. Moved dry run check. Now, will just log and not execute the pipeline.